### PR TITLE
BUG: fix custom xticks in bar() plotting function

### LINKED ISSUE #14119: unable to set xticks within bar() plotting function
#### Code Sample, a copy-pastable example if possible

In the code below:

``` python
pd.Series(range(10), index=range(10)).plot.bar(xticks=range(0, 11, 2))
```

`xticks` argument has no effect.  However, there's no such problem in line() plot.
#### Expected Output

Output plot is expected to have xticks defined by `xticks` argument, e.g. in the example above it should be set to [0, 2, 4, 6, 8, 10].
#### output of `pd.show_versions()`

commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.1
nose: None
pip: 8.1.2
setuptools: 23.0.0
Cython: None
numpy: 1.11.1
scipy: 0.17.1
statsmodels: None
xarray: 0.8.1
IPython: 5.0.0
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.4
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None

### PR DESCRIPTION
**Pull request by GiuPassarelli**
- [x] closes #14119
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by TomAugspurger — COMMENTED** [review]
Thanks! Can you add the example from https://github.com/pandas-dev/pandas/issues/14119 as a test in pandas/tests/plotting/test_series.py` and make an assertion that the output x tick labels match the expected?

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -171,6 +171,7 @@ Plotting
 - Bug in :meth:`DataFrame.plot` producing incorrect legend markers when plotting multiple series on the same axis (:issue:`18222`)
 - Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datetime or timedelta data. These types are now automatically dropped (:issue:`22799`)
 - Bug in :meth:`DataFrame.plot.line` and :meth:`DataFrame.plot.area` produce wrong xlim in x-axis (:issue:`27686`, :issue:`25160`, :issue:`24784`)
+- Bug in :core.py:`BarPlot._decorate_ticks` xticks argument has no effect in bar plot (:issue:`14119`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.0.0.rst]: We shouldn't refer to private methods in the release notes, which are for the user. Perhaps something like

```suggestion
- Bug in the ``xticks`` argument being ignored for :meth:`DataFrame.plot.bar` (:issue:`14119`)
```

File: pandas/plotting/_matplotlib/core.py [hunk 2]
```diff
@@ -1435,8 +1435,13 @@ def _post_plot_logic(self, ax, data):
 
     def _decorate_ticks(self, ax, name, ticklabels, start_edge, end_edge):
         ax.set_xlim((start_edge, end_edge))
-        ax.set_xticks(self.tick_pos)
-        ax.set_xticklabels(ticklabels)
+
+        if self.xticks is not None:
+            ax.set_xticks(np.array(self.xticks))
```
Comment (TomAugspurger) [inline pandas/plotting/_matplotlib/core.py]: Just to verify, ignoring `ticklabels` is the correct thing to do here?

File: pandas/tests/plotting/test_series.py [hunk 3]
```diff
@@ -871,6 +871,13 @@ def test_xticklabels(self):
         exp = ["P{i:02d}".format(i=i) for i in [0, 3, 5, 9]]
         self._check_text_labels(ax.get_xticklabels(), exp)
 
+    def test_xtick_barPlot(self):
+        # GH28172
+        s = pd.Series(range(10), index=["P{i:02d}".format(i=i) for i in range(10)])
+        ax = s.plot.bar(xticks=range(0, 11, 2))
+        exp = np.array(list(range(0, 11, 2)))
+        assert (exp == ax.get_xticks()).all()
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_series.py]: Write this as `tm.assert_numpy_array_equal(exp, ax.get_xticks())`.

**Review by TomAugspurger — COMMENTED** [review]
Looks really close, just a couple small comments.

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (GiuPassarelli) [reply pandas/plotting/_matplotlib/core.py]: I don't know, the ticklabels doesn't work properly when the index has a different size than the axis. If I use it in this case the x axis only goes up to 5. What is the best way to go about it?

Comment (TomAugspurger) [reply pandas/plotting/_matplotlib/core.py]: Sorry, I'm not sure I understand the issue. To be clear, my question wasn't saying that you shouldn't ignore `ticklabels`. That may be the correct thing to do (for example, when the number of user-provided `xticks` differs from the number of original points.

> If I use it in this case the x axis only goes up to 5. 

Is "this case" your new test case?

**Review by TomAugspurger — APPROVED** [review]
This looks good, pending the discussion in https://github.com/pandas-dev/pandas/pull/28172#discussion_r319637757

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (TomAugspurger) [reply pandas/plotting/_matplotlib/core.py]: I think this is the correct behavior.

**Review by TomAugspurger — APPROVED** [review]
LGTM. Ping on green.

**Review by WillAyd — APPROVED** [review]
lgtm merging on green

**Comment by TomAugspurger** [comment]
Thanks @GiuPassarelli!

**Pull request merged**